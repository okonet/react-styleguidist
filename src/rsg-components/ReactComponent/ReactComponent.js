import React, { PropTypes } from 'react';
import Markdown from 'rsg-components/Markdown';
import Props from 'rsg-components/Props';
import Methods from 'rsg-components/Methods';
import Examples from 'rsg-components/Examples';
import ReactComponentRenderer from 'rsg-components/ReactComponent/ReactComponentRenderer';

export default function ReactComponent({ component }, { isolatedComponent = false }) {
	const { name, pathLine, examples } = component;
	const { description, props, methods } = component.props;
	return (
		<ReactComponentRenderer
			name={name}
			pathLine={pathLine}
			description={description && <Markdown text={description} />}
			props={props && <Props props={props} />}
			methods={methods.length > 0 && <Methods methods={methods} />}
			examples={examples && <Examples examples={examples} name={name} />}
			isolated={isolatedComponent}
		/>
	);
}

ReactComponent.propTypes = {
	component: PropTypes.object.isRequired,
};

ReactComponent.contextTypes = {
	isolatedComponent: PropTypes.bool,
};
