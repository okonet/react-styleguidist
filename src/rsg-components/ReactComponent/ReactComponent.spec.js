import React from 'react';
import noop from 'lodash/noop';
import Markdown from '../Markdown';
import Playground from '../Playground';
import ReactComponent from './ReactComponent';
import { ReactComponentRenderer } from './ReactComponentRenderer';

const component = {
	name: 'Foo',
	pathLine: 'foo/bar.js',
	props: {
		description: 'Bar',
	},
	examples: [
		{
			type: 'code',
			content: '<button>OK</button>',
			evalInContext: noop,
		},
		{
			type: 'markdown',
			content: 'Hello *world*!',
		},
	],
};

it('should render component renderer', () => {
	const actual = shallow(
		<ReactComponent
			component={component}
		/>
	);

	expect(actual).toMatchSnapshot();
});

it('renderer should render component', () => {
	const actual = shallow(
		<ReactComponentRenderer
			classes={{}}
			name={component.name}
			pathLine={component.pathLine}
			description={component.props.description}
			examples={[
				<Playground
					key={0}
					index={0}
					name="Component"
					code={component.examples[0].content}
					evalInContext={component.examples[0].evalInContext}
				/>,
				<Markdown
					key={1}
					text={component.examples[1].content}
				/>,
			]}
		/>
	);

	expect(actual).toMatchSnapshot();
});

test('should render props section', () => {
	const actual = shallow(
		<ReactComponentRenderer
			classes={{}}
			name="Test"
			pathLine="test"
			props={<div>test</div>}
		/>
	);

	expect(actual).toMatchSnapshot();
});

test('should render methods section', () => {
	const actual = shallow(
		<ReactComponentRenderer
			classes={{}}
			name="Test"
			pathLine="test"
			props={null}
			methods={<div>test</div>}
		/>
	);

	expect(actual).toMatchSnapshot();
});

test('should render both props and methods section', () => {
	const actual = shallow(
		<ReactComponentRenderer
			classes={{}}
			name="Test"
			pathLine="test"
			props={<div>prop</div>}
			methods={<div>method</div>}
		/>
	);

	expect(actual).toMatchSnapshot();
});

test('should not render props / methods section if there is no content', () => {
	const actual = shallow(
		<ReactComponentRenderer
			classes={{}}
			name="Test"
			pathLine="test"
		/>
	);

	expect(actual).toMatchSnapshot();
});
