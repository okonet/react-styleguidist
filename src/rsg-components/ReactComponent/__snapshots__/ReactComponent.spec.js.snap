exports[`test renderer should render component 1`] = `
<div
  id="Foo-container">
  <header>
    <h2
      id="Foo">
      Foo
    </h2>
    <div>
      foo/bar.js
    </div>
    <div>
      <_class
        href="/">
        ← Back
      </_class>
    </div>
  </header>
  <div>
    Bar
  </div>
  <Playground
    code="<button>OK</button>"
    evalInContext={[Function]}
    index={0}
    name="Component" />
  <_class
    text="Hello *world*!" />
</div>
`;

exports[`test should not render props / methods section if there is no content 1`] = `
<div
  id="Test-container">
  <header>
    <h2
      id="Test">
      Test
    </h2>
    <div>
      test
    </div>
    <div>
      <_class
        href="/">
        ← Back
      </_class>
    </div>
  </header>
  <div />
</div>
`;

exports[`test should render both props and methods section 1`] = `
<div
  id="Test-container">
  <header>
    <h2
      id="Test">
      Test
    </h2>
    <div>
      test
    </div>
    <div>
      <_class
        href="/">
        ← Back
      </_class>
    </div>
  </header>
  <div />
  <div>
    <h3>
      Props
    </h3>
    <div>
      prop
    </div>
  </div>
  <div>
    <h3>
      Methods
    </h3>
    <div>
      method
    </div>
  </div>
</div>
`;

exports[`test should render component renderer 1`] = `
<_class
  description={
    <_class
      text="Bar" />
  }
  examples={
    <Examples
      examples={
        Array [
          Object {
            "content": "<button>OK</button>",
            "evalInContext": [Function],
            "type": "code",
          },
          Object {
            "content": "Hello *world*!",
            "type": "markdown",
          },
        ]
      }
      name="Foo" />
  }
  name="Foo"
  pathLine="foo/bar.js" />
`;

exports[`test should render methods section 1`] = `
<div
  id="Test-container">
  <header>
    <h2
      id="Test">
      Test
    </h2>
    <div>
      test
    </div>
    <div>
      <_class
        href="/">
        ← Back
      </_class>
    </div>
  </header>
  <div />
  <div>
    <h3>
      Methods
    </h3>
    <div>
      test
    </div>
  </div>
</div>
`;

exports[`test should render props section 1`] = `
<div
  id="Test-container">
  <header>
    <h2
      id="Test">
      Test
    </h2>
    <div>
      test
    </div>
    <div>
      <_class
        href="/">
        ← Back
      </_class>
    </div>
  </header>
  <div />
  <div>
    <h3>
      Props
    </h3>
    <div>
      test
    </div>
  </div>
</div>
`;
